File: ComInLanDroid/cominlan/src/main/java/com/onballgroup/cominlan/model/CServer.java
package com.onballgroup.cominlan.model;

import com.onballgroup.cominlan.model.Base.BaseModel;

import java.net.InetAddress;

/**
 * Created by Phong Le on 4/20/2016.
 */
public class CServer extends BaseModel implements IServer {
    //--------------Created from packet------------------//
    private String _id;
    private String _name;
    private int _port;

    public CServer() {
        refresh();
        _state = ServerState.None;
    }

    @Override
    public String getId() {
        return _id;
    }

    public void setId(String id) {
        _id = id;
    }

    @Override
    public String getName() {
        return _name;
    }

    public void setName(String name) {
        _name = name;
    }

    @Override
    public int getPort() {
        return _port;
    }

    public void setPort(int port) {
        _port = port;
    }

    //--------------Created by app------------------//
    private InetAddress _address;
    private ServerState _state;
    private String _checksum;
    private long _refreshTime;

    @Override
    public InetAddress getAddress() {
        return _address;
    }

    public void setAddress(InetAddress address) {
        _address = address;
    }

    @Override
    public ServerState getState() {
        return _state;
    }

    public void setState(ServerState state) {
        _state = state;

        if (_onServerListener != null)
        {
            _onServerListener.onStateChanged(this);
        }
    }

    @Override
    public String getChecksum() {
        return _checksum;
    }

    @Override
    public long getRefreshTime() {
        return _refreshTime;
    }

    //--------------Methods------------------//
    public void calculateChecksum() {
        _checksum = calculateChecksum(_id + _name);
    }

    public void refresh() {
        _refreshTime = getCurrentUnixTimestamp();
    }

    //--------------Events------------------//
    private OnServerListener _onServerListener;

    @Override
    public void setOnServerListener(OnServerListener listener) {
        _onServerListener = listener;
    }

    public void callIDataReceived(String data)
    {
        if (_onServerListener != null)
        {
            _onServerListener.onDataReceived(this, data);
        }
    }
}

File: ComInLanWindows/ComInLan/Model/CServer.cs
using ComInLan.Model.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;

namespace ComInLan.Model
{
	public class CServer : BaseModel, IServer
	{
		//--------------Created from packet------------------//
		public string Id { get; set; }

		public string Name { get; set; }

		public int Port { get; set; }

		public CServer()
		{
			Refresh();
			State = ServerState.None;
		}

		//--------------Created by app------------------//
		public IPAddress Address { get; set; }

		private ServerState _state;
		public ServerState State
		{
			get { return _state; }
			set
			{
				_state = value;

				if (StateChanged != null)
				{
					StateChanged(this);
				}
			}
		}

		public string Checksum { get; private set; }

		public long RefreshTime { get; private set; }


		//--------------Methods------------------//
		public void CalculateChecksum()
		{
			Checksum = CalculateChecksum(Id + Name);
		}

		public void Refresh()
		{
			RefreshTime = GetCurrentUnixTimestamp();
		}

		//--------------Events------------------//
		public event ServerStateEventHandler StateChanged;
		public event SeverDataEventHandler DataReceived;

		public void CallIDataReceived(String data)
		{
			if (DataReceived != null)
			{
				DataReceived(this, data);
			}
		}
	}
}
